determined-ai/determined
# chore: Pinned workspaces in stores/workspaces [WEB-586]

## Description

Incorporates filters into `useWorkspaces` so that it can take the place of `useFetchPinnedWorkspaces`. This includes shifting the internal state from `workspaces` (the `Loadable<Workspace[]>`) to a workspacesIndex `Map<string, Loadable<Workspace[]>>`

Work in progress while addressing the experiment store PR (confirming this use of Immutable Map), and with Move Experiment modal.

## Test Plan

- Pin at least one workspace
- Load Determined and confirm that the pinned workspace is shown on the left navigation
- When you refresh the page, the left navigation briefly shows a circle spinner before loading pinned workspaces
- Unpin the workspace and confirm that it disappears from the left navigation within a few seconds
- On the Workspaces view, each workspace has a card. Use the menu on the card to pin and unpin a workspace. The menu label should show 'pin workspace' and 'unpin workspace'
- Open an experiment. Select another workspace and project to move it.

## Checklist
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
